ESP32main.ino

```diff
@@ -21,6 +21,8 @@ const int VOLUME    = 1;
 const int DATA      = 2;
 
 volatile int state=NODATA;
+SemaphoreHandle_t dataSemaphore = NULL;
+String data = "";
```

Comment:
dataSemaphore and data are shared between the microphone threads and client thread. If you need more variables that can be concurrent-sensitive please add here @olgaRuiz7